### PR TITLE
[fix]add nodeSelector design doc

### PR DESCRIPTION
**Pull request by qiankunli**
Signed-off-by: bert.li <qiankun.li@qq.com>
design doc for #1821

### CODE REVIEW

File: docs/design/node-selector.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+## Introduction
+
+for example, my k8s cluster has 10 nodes (10 gpu per node), 5 for training, 5 for serving, I want to use volcano schedule tfjob/pytorchjob/vcjob on training nodes. use default-scheduler schedule online pod on serving nodes. it is work by setting nodeSelector or nodeAffinity to pod, but it is not properly when considering volcano queue mechanism.
+
+if there are two queue: queue1(weight=1) and queue2 (weight=1), it is overused if queue1 used 40 gpu, but it's weight resource is 50 (100 * 1/2), so if queue2 want use 20 gpu, reclaimable mechanism will not work.
```
Comment (Thor-wl) [inline docs/design/node-selector.md:5]: If there are totally 100 GPUs in the cluster, the deserved resource for queue1 and queue2 are both 50 GPUS. Why is it overused if queue1 used 40 GPUs? BTW, please make the format and grammar of the doc more canonical.

File: docs/design/node-selector.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+so it is necessary to limit volcano scheduler that it can only work on training nodes, not all nodes in cluster.
+
+so I add nodeSelector for volcano scheduler.
+
+## Usage
+
+in following example, volcano can only work on the node which has `node-role:training` or `node-role:dev` or `gpu-model: tesla` label. you can use any label name you like.
+
+```yaml
```
Comment (Thor-wl) [inline docs/design/node-selector.md:15]: Sugeest to add comment about the configuration to point out what should be foucsed on.

File: docs/design/node-selector.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      - name: conformance
+    - plugins:
+      - name: overcommit
+      - name: drf
+      - name: predicates
+      - name: proportion
+      - name: nodeorder
+      - name: binpack
+    nodeSelector:
+      node-role: training,dev
```
Comment (Thor-wl) [inline docs/design/node-selector.md:30]: Pls separate `node-role` to different items

File: docs/design/node-selector.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      gpu-model: tesla
+```
+
+## Design
+
+volcano use nodeInformer to get all node list in k8s cluster. so we can filter the node list by nodeSelector label.
```
Comment (Thor-wl) [inline docs/design/node-selector.md:36]: This is implementation part and suggest to give more details about what should be done clearly.

**Review by shinytang6 — CHANGES_REQUESTED** [review]
Two suggestions:
1. pods should also be filtered by specified nodes
2. I recommend putting the configuration of node-selector in the scheduler parameters

**Comment by Thor-wl** [comment]
> Two suggestions:
> 
> 1. pods should also be filtered by specified nodes
> 2. I recommend putting the configuration of node-selector in the scheduler parameters

If node-selector is put as scheduer parameters, may it be not so flexible when make modifications?

**Comment by shinytang6** [comment]
> > Two suggestions:
> > 
> > 1. pods should also be filtered by specified nodes
> > 2. I recommend putting the configuration of node-selector in the scheduler parameters
> 
> If node-selector is put as scheduer parameters, may it be not so flexible when make modifications?

The configuration file is hot updated, but it will not resync after the update, which may cause some of the changed nodes not be added to the cache

**Comment by qiankunli** [comment]
> > > Two suggestions:
> > > 
> > > 1. pods should also be filtered by specified nodes
> > > 2. I recommend putting the configuration of node-selector in the scheduler parameters
> > 
> > 
> > If node-selector is put as scheduer parameters, may it be not so flexible when make modifications?
> 
> The configuration file is hot updated, but it will not resync after the update, which may cause some of the changed nodes not be added to the cache
I have changed the `schedule.conf` file watch code to watch the changed nodeSelector config

**Comment by qiankunli** [comment]
I have changed the schedule.conf file watch code to watch the changed nodeSelector config

**Comment by Thor-wl** [comment]
> > > Two suggestions:
> > > 
> > > 1. pods should also be filtered by specified nodes
> > > 2. I recommend putting the configuration of node-selector in the scheduler parameters
> > 
> > 
> > If node-selector is put as scheduer parameters, may it be not so flexible when make modifications?
> 
> The configuration file is hot updated, but it will not resync after the update, which may cause some of the changed nodes not be added to the cache

Well, yes, that may be a problem.

**Review by shinytang6 — COMMENTED** [review]
lgtm overall

**Review by jasonliu747 — COMMENTED** [review]
lgtm overall, except some grammar issues.
I'll give more comments on Monday~

**Pull request merged**